src/Keboola/S3Extractor/Finder.php

```diff
@@ -60,130 +60,159 @@ public function __construct(Config $config, State $state, LoggerInterface $logge
      */
     public function findFiles(): array
     {
-        /** @var File[] */
-        $filesToDownload = [];
-
         $this->logger->info('Listing files to be downloaded');
 
         // Detect wildcard at the end
         if (substr($this->key, -1) == '*') {
-            $paginator = $this->client->getPaginator(
-                'ListObjectsV2',
-                [
-                    'Bucket' => $this->bucket,
-                    'Prefix' => substr($this->key, 0, -1),
```

Comment:
Vsetko bolo v jednej metode, podelil som to do viacerych metod, aby bolo jasne, co sa tam deje.